crates/rattler/src/install/link.rs

```diff
@@ -422,9 +451,7 @@ fn hardlink_to_destination(
         match fs::hard_link(source_path, destination_path) {
             Ok(_) => return Ok(LinkMethod::Hardlink),
             Err(e) if e.kind() == ErrorKind::AlreadyExists => {
-                fs::remove_file(destination_path).map_err(|err| {
-                    LinkFileError::IoError(String::from("removing clobbered file"), err)
-                })?;
+                remove_file_or_empty_dirs(destination_path)?;
```

Comment:
I think this could cause a race condition. During linking of a single file we recursively construct directories and then install a file into it. However, this operation is not atomic so it could happen that before installing a file to a directory the directory is accidentally deleted by the removal of another file (and its directories).